core-bundle/src/Search/Document.php

```diff
@@ -244,4 +244,22 @@ private function filterJsonLdContexts(array $data, array $contexts): array
 
         return $found ? $newData : [];
     }
+
+    public function __serialize(): array
```

Comment:
Required because Symfony Messenger serializes objects by default for the transport. Without explicit methods, PHP would try to serialize `DOM...` objects which is not supported.